examples/lock-app/nrfconnect/main/AppTask.cpp

```diff
@@ -549,11 +550,18 @@ void AppTask::DispatchEvent(AppEvent * aEvent)
 
 void AppTask::UpdateClusterState()
 {
-    // write the new on/off value
-    EmberAfStatus status = Clusters::OnOff::Attributes::OnOff::Set(kLockEndpointId, !BoltLockMgr().IsUnlocked());
-
+    EmberAfStatus status;
+    LOG_INF("Updating door lock state");
+    chip::app::DataModel::Nullable<DlLockState> lockSate;
+    status = Clusters::DoorLock::Attributes::LockState::Get(kLockEndpointId, lockSate);
```

Comment:
The purpose of this function was to synchronize the cluster state with `BoltLockMgr().IsUnlock()`. The latter could be manipulated with a Button. This code, however, seems to toggle the state, which I'm not sure is 100% accurate.

Reply:
You make a good point. I was running into issue, where the BoltLockMgr state would not match the server state and the button would not change the state. I will update this PR to use the callbacks below to update BoltLockMgr state instead.

Reply:
Updated and retested.